src/app/games/diaballik/diaballik.component.ts

```diff
@@ -67,29 +70,49 @@ export class DiaballikComponent
         this.board = state.board; // Needed by RectangularGameComponent
         this.stateInConstruction = state;
         const possibleVictory: MGPOptional<VictoryOrDefeatCoords> = this.rules.getVictoryOrDefeatCoords(state);
+        this.victoryCoord = MGPOptional.empty();
+        this.loserCoords = [];
+        this.blockedLines = [];
         if (possibleVictory.isPresent()) {
             const victory: VictoryOrDefeatCoords = possibleVictory.get();
             if (victory instanceof VictoryCoord) {
                 this.victoryCoord = MGPOptional.of(victory.coord);
             } else if (victory instanceof DefeatCoords) {
-                this.defeatCoords = victory.coords;
+                this.loserCoords = victory.allLoserPieces;
+                // If Player.ZERO won, the line will be placed in front of their piece directly.
+                // If Player.ONE won, we need to shift them by one space size so that they appear in front of the piece
+                const shift: number = victory.winner === Player.ZERO ? 0 : this.SPACE_SIZE;
+                for (const blockedPiece of victory.opponentPiecesInContact) {
+                    const x: number = blockedPiece.x * this.SPACE_SIZE;
+                    const y: number = blockedPiece.y * this.SPACE_SIZE + shift;
+                    const line: Line = new Line(x, y, x + this.SPACE_SIZE, y);
+                    this.blockedLines.push(line);
+                }
             }
         }
     }
 
     public override async showLastMove(move: DiaballikMove): Promise<void> {
-        this.lastMovedPiecesCoords = [];
-        this.lastMovedBallCoords = [];
+        this.hideLastMove();
```

Comment:
je crois que c'est un lifecycle sin ça !
C'est nécessaire ?

Reply:
j'enlève et on verra au prochain test !

Reply:
TODO: à investiguer, si après un coup adverse, je fais un mouvement, puis je commence un deuxième et j'annule, je vois toujours mon premier sous-mouvement alors qu'il n'est plus fait

Reply:
même pas besoin de commencer le second, juste faire un submove puis annuler

Reply:
hideLastMove réajouté du coup !

Reply:
Quand cancelMove est appellé sur un GameComponent -> showLastMove s'en suit et avant eux cancelMoveAttempt et cancelMoveOnWrapper sont appelés, le bug dont je me souvient dans Diaballik était lié à la move attempt qui était montré !
Donc enlève cette hérésie dont aucun autres jeu n'as besoin (et certains sont plus chaud et malsainement faits que diaballik) et ajotue cancelMoveAttempt

Reply:
ah mais oui, hideLastMove n'était donc pas nécessaire ! 
Mais donc hideLastMove est inutile ? Ici hideLastMove fait litérallement plus rien et tout fonctionne !

Reply:
Re: mouais en fait ici il est utilisé par le composant diaballik lui-même pour cacher le lastMove quand on commence un nouveau move. Mais dans quel cas hideLastMove est appelé par quelqu'un d'autre ?

Reply:
Dans GameWrapper
```
protected async updateBoardAndShowLastMove(triggerAnimation: boolean): Promise<void> {
        this.gameComponent.cancelMoveAttempt();
        this.gameComponent.hideLastMove();
        await this.gameComponent.updateBoard(triggerAnimation);
        if (this.gameComponent.node.previousMove.isPresent()) {
            const move: Move = this.gameComponent.node.previousMove.get();
            await this.gameComponent.showLastMove(move);
        }
    }
    ```
LGWC.restartGame l'appelle avant d'updater le board
10 jeux hérétiques l'appellent encore manuellement, mais c'est idéalement déconseillé par la kodemandantur !

Reply:
ok! Résolvable ou y'a une action à faire ?
Est-ce que cacher manuellement le last-move ici n'est pas la seule bonne chose à faire ?

Reply:
cacher le lastMove est en effet une action ok quand tu commence à créer du mouvement multi-clic, et le Wrapper peut pas savoir que tu es un multiclic ni que tu as commencé un mouvement. donc ces dix jeux sont peut être pas hérétique mais juste dans ce cas là !

Mais içi quand tu crée du move tu REMPLACE le highlightedCoords par celles du oingoing move, cachant de-facto le lastMove, donc pas de modif à faire là, si ?

Tiens en fait, clicValidity pourrait dire "si le clic est légal: l'user est interactif, on hideLastMove car il en crée un" non ??? ça éviterais l'exception mentionnée ci-dessus (à sub-ticketter ?) non ??

Reply:
Ici tu remplaces pas, tu ajoute des highlightedCoord au fur et à mesure que tu construit. La création du premier sous mouvement appelle donc hideLastMove pour clearer (et donc "remplacer").

Que `canUserPlay` appelle du `hideLastMove` pourquoi pas, mais faudrait adapter son nom !
J'te propose le ticket: [W] [easy] GameComponent: canUserPlay should call hideLastMove when user is interactive (and be renamed in accordance with that behavior)